packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -546,51 +675,136 @@ abstract class _AnimatedScrollViewState<T extends _AnimatedScrollView> extends S
   /// to [AnimatedGrid.itemBuilder] or [AnimatedList.itemBuilder] when the item
   /// is visible.
   ///
+  /// If using [AnimatedList.separated] the animation will also be passed
+  /// to `separatorBuilder`.
+  ///
   /// This method's semantics are the same as Dart's [List.insert] method: it
   /// increases the length of the list of items by one and shifts
   /// all items at or after [index] towards the end of the list of items.
   void insertItem(int index, { Duration duration = _kDuration }) {
-    _sliverAnimatedMultiBoxKey.currentState!.insertItem(index, duration: duration);
+    if (widget.removedSeparatorBuilder == null) {
+      _sliverAnimatedMultiBoxKey.currentState!.insertItem(index, duration: duration);
+    } else {
+      final int itemIndex = _computeItemIndex(index);
+      _sliverAnimatedMultiBoxKey.currentState!.insertItem(itemIndex, duration: duration);
+      if (_itemsCount > 1) {
+        _sliverAnimatedMultiBoxKey.currentState!.insertItem(itemIndex, duration: duration);
```

Comment:
I think this is adding the separator, can you add a comment if so?
What if this is the last item in the list, does it still build a separator?

Reply:
I added a comment. Does this help? Will be happy to improve this further if needed!